crates/astria-merkle/src/audit.rs

```diff
@@ -621,4 +621,21 @@ impl Proof {
     pub fn verify(&self, leaf: &[u8], root_hash: [u8; 32]) -> bool {
         self.audit().with_leaf(leaf).with_root(root_hash).perform()
     }
+
+    /// This should only be used where `parts` has been provided by a trusted entity, e.g. read from
+    /// our own state store.
+    #[doc(hidden)]
+    #[must_use]
+    pub fn unchecked_from_parts(parts: UncheckedProof) -> Self {
```

Comment:
Can you feature gate this one as well?

Reply:
Done in [ccc3f9e](https://github.com/astriaorg/astria/pull/1492/commits/ccc3f9e62661d8c6f0d73a52a119ed18aa89cc60).